realgud/common/cmds.el

```diff
@@ -291,7 +291,14 @@ EVENT should be a mouse click on the left fringe or margin."
 (defun realgud:cmd-eval-at-point()
   "Eval symbol under point."
   (interactive)
-  (realgud:cmd-run-command (thing-at-point 'symbol) "eval"))
+
+  (beginning-of-thing 'symbol)
+  (set-mark-command 'nil)
+  (end-of-thing 'symbol)
+
+  (realgud:cmd-run-command
+   (read-string "Eval: " (thing-at-point 'symbol))
```

Comment:
I like the fact that something is marked. It would be neat if the completion at this point would just add the next word so that one could hit tab and fill out more and more of the text.

Also, not related to this change, I notice that `realgud:cmd-eval-dwim` will lose a marked region after evaluation. Is it wrapping `call-cmd-interactively` in a  `save-excursion` that will preserve the region?

Reply:
Sigh. save-excursion and save-mark-and-excursion all have no effect since the point and mark haven't changed. It is apparently a  highlight transient mode bug. See https://stackoverflow.com/questions/30243315/marked-region-does-not-highlight-in-emacs-after-changing-the-buffer

Reply:
tab grabbing the next thing-at-point does seem like a good idea.  i'm not sure how you could catch the tab key in read-string though, seems like read-string is a synchronous function so i'm not sure its possible.

i also want to address the bug at https://github.com/realgud/realgud/issues/180 and maybe something like this https://github.com/realgud/realgud/issues/191 so not sure what my priorities are now.

Reply:
> I'm not sure how you could catch the tab key in read-string though, seems like read-string is a synchronous function so i'm not sure its possible.

I think one would use [`completing-read` and `completing-read-function`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Minibuffer-Completion.html).